lib/bundler/uri_credentials_filter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Bundler
+  module URICredentialsFilter
+    def anonymized_uri(uri_to_anonymize)
+      return uri_to_anonymize if uri_to_anonymize.nil?
+      uri = uri_to_anonymize.dup
+      uri = URI(uri.to_s) unless uri.is_a?(URI)
+      uri.user = uri.password = nil if uri.userinfo
```

Comment:
Hm, okay, fair enough. Any suggestions on how to handle the `oauth_token:x-oauth-basic@` format for authentication and removing the `oauth_token` part? Removing just the password on this credential would render it `oauth_token@` instead of `x-oauth-basic@` as the `URI` module detects `oauth_token` as the user. 

Is `oauth_token:x-oauth-basic@` a valid authentication usage/format for git sources for Bundler? I don't have the first-hand context on this and am basing this oauth example off of the discussion at bundler/bundler-features#111

If `x-oauth-basic` will always be the "URI password" part of the uri for oauth authentication, then I could handle both regular login credentials and oauth credentials. I'm seeing a lot of sources that support the idea that this is the right assumption, though I'm not 100% sure.